doc(home-plugin): Adds documentation on <HomePageVisitedByType/> usage (#19688)
backstage/backstage: plugins/home/README.md

<CustomHomepageGrid config={defaultConfig}>
```

## Page visit homepage component (HomePageVisitedByType)

> This component shows the homepage user a view for "Recently visited" or "Top visited".

Reviewer: Maybe clarify what a visit is. Is that a page visit?

> Being provided by the `<HomePageVisitedByType/>` component, see it in usage on a homepage example bellow:

Reviewer 2: ```suggestion
Being provided by the `<HomePageVisitedByType/>` component, see it in use on a homepage example below:
```

```tsx

> import React from 'react';

Reviewer: ```suggestion
// packages/app/src/components/home/HomePage.tsx
import React from 'react';
```

import Grid from '@material-ui/core/Grid';
import { HomePageVisitedByType } from '@backstage/plugin-home';

export const homePage = (
  <Grid container spacing={3}>
    <Grid item xs={12} md={4}>
      <HomePageVisitedByType kind="top" />
    </Grid>
    <Grid item xs={12} md={4}>
      <HomePageVisitedByType kind="recent" />
    </Grid>
  </Grid>
);
```

> There ase some requirements to provide its functionality, so please ensure the following:

Reviewer 2: how did this PR pass spell check? :) something fishy

```suggestion
There are some requirements to provide its functionality, so please ensure the following:
```

Reviewer 2: oh; since you aren't targeting master, i think that's why!

> These components need an api to handle visit data, please refer to the [utility-apis](../../docs/api/utility-apis.md)

Reviewer 2: ```suggestion
These components need an API to handle visit data, please refer to the [utility-apis](../../docs/api/utility-apis.md)
```

documentation for more information. Bellow you can see an example for two options:

```ts

> // apis.ts

Reviewer: ```suggestion
// packages/app/src/apis.ts
```

// ...
import {
  CoreStorageVisitsApi,
  LocalStoreVisitsApi,
  visitsApiRef,
} from '@backstage/plugin-home';
// ...
export const apis: AnyApiFactory[] = [
  // Implementation that relies on the integration with storageApi
  createApiFactory({
    api: visitsApiRef,
    deps: {
      storageApi: storageApiRef,
      identityApi: identityApiRef,
    },
    factory: ({ storageApi, identityApi }) =>
      CoreStorageVisitsApi.create({ storageApi, identityApi }),
  }),

  // Or a local data implementation, relies on the browser's window.localStorage
  createApiFactory({
    api: visitsApiRef,
    deps: {
      identityApi: identityApiRef,
    },
    factory: ({ identityApi }) => LocalStoreVisitsApi.create({ identityApi }),
  }),
  // ...
```

To monitor page visit activity and save it on behalf of the user a component is provided, please add it to your app.
See the example usage:

```ts

> // App.tsx

Reviewer: ```suggestion
// packages/app/src/App.tsx
```
